agent_build_refactored/managed_packages/managed_packages_builders.py

```diff
@@ -972,269 +1082,477 @@ def create_build_dependencies_step(
+    """
+
+    steps = {}
+    for architecture in SUPPORTED_ARCHITECTURES:
+        run_in_remote_docker = architecture != Architecture.X86_64
+
+        step = ArtifactRunnerStep(
+            name=f"build_python_dependencies_{architecture.value}",
+            script_path="agent_build_refactored/managed_packages/steps/build_python_dependencies.sh",
+            base=INSTALL_BUILD_ENVIRONMENT_STEPS[architecture],
+            required_steps={
+                "DOWNLOAD_BUILD_DEPENDENCIES": DOWNLOAD_PYTHON_DEPENDENCIES
+            },
+            environment_variables={
+                **_PYTHON_BUILD_DEPENDENCIES_VERSIONS,
+            },
+            github_actions_settings=GitHubActionsSettings(
+                run_in_remote_docker=run_in_remote_docker,
+                cacheable=True
+            )
         )
-    )
+        steps[architecture] = step
+
+    return steps
 
 
-def create_build_agent_libs_step(
-        base_step: EnvironmentRunnerStep,
-        build_python_step: ArtifactRunnerStep,
-        run_in_remote_docker: bool = False
-):
+def create_build_python_steps(
+        openssl_version_type: str
+) -> Dict[Architecture, ArtifactRunnerStep]:
     """
-    Function that creates step that installs agent requirement libraries.
-    :param base_step: Step with environment where to build.
-    :param build_python_step: Required step that builds Python.
-    :param run_in_remote_docker: If possible, run this step in remote docker engine.
-    :return: Result step.
+    Function that creates step instances that build Python interpreter.
+    :return: Result steps mapped to architectures..
     """
 
-    return ArtifactRunnerStep(
-        name=f"build_agent_libs_{base_step.architecture.value}",
-        script_path="agent_build_refactored/managed_packages/steps/build_agent_libs.sh",
-        tracked_files_globs=[
-            "agent_build/requirement-files/*.txt",
-            "dev-requirements.txt"
-        ],
-        base=base_step,
-        required_steps={
-            "BUILD_PYTHON": build_python_step
-        },
-        environment_variables={
-            "PYTHON_SHORT_VERSION": EMBEDDED_PYTHON_SHORT_VERSION,
-            "RUST_VERSION": "1.63.0",
-            "SUBDIR_NAME": AGENT_DEPENDENCY_PACKAGE_SUBDIR_NAME
-        },
-        github_actions_settings=GitHubActionsSettings(
-            cacheable=True,
-            run_in_remote_docker=run_in_remote_docker
+    install_prefix = f"/opt/{AGENT_DEPENDENCY_PACKAGE_SUBDIR_NAME}"
+
+    steps = {}
+
+    for architecture in SUPPORTED_ARCHITECTURES:
+        run_in_remote_docker = architecture != Architecture.X86_64
+
+        build_python = ArtifactRunnerStep(
+            name=f"build_python_{openssl_version_type}_{architecture.value}",
+            script_path="agent_build_refactored/managed_packages/steps/build_python.sh",
+            base=INSTALL_BUILD_ENVIRONMENT_STEPS[architecture],
+            required_steps={
```

Comment:
Since this value ``required_steps`` is a dictionary which is unordered by default, how do we achieve right order of those steps? 

Parent class (``RunnerStep``) description also says ":param required_steps: List of other steps that has to be executed in order to run this step." which is a bit confusing.

Reply:
Sorry, that docstring was left outdated. The `required_steps` is used by step in order to access result artifacts of the `required_steps`, order here is irrelevant.